alpenhorn/storage.py

```diff
@@ -1,18 +1,75 @@
 """StorageNode and StorageGroup table models."""
 # Type annotation shennanigans
 from __future__ import annotations
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, overload
+
+import importlib
+import peewee as pw
+from peewee import fn
+
+from .db import EnumField, base_model
+from . import util
 
 if TYPE_CHECKING:
     import pathlib
     from .acquisition import ArchiveFile
+    from .io.base import BaseNodeIO, BaseGroupIO
 
 
-import peewee as pw
-from peewee import fn
+@overload
+def _get_io_instance(obj: StorageNode) -> BaseNodeIO:
+    ...
 
-from .db import EnumField, base_model
-from . import util
+
+@overload
+def _get_io_instance(obj: StorageGroup) -> BaseGroupIO:
+    ...
+
+
```

Comment:
```suggestion
```

Reply:
These lines are added by `black`.  We can't remove them.

Reply:
Oh, interesting. My bad